### PR TITLE
test: add test for validating numeric values into system contract functions

### LINKED ISSUE #14312: HTS Non-static simple input functions
These are the functions to consider for this task.

burnToken(address,uint64,int64[])
burnToken(address,int64,int64[])
approve(address,uint256)
approve(address,uint256)
approve(address,address,uint256)
approveNFT(address,address,uint256)
mintToken(address,uint64,bytes[])
mintToken(address,int64,bytes[])
wipeTokenAccount(address,address,uint32)
wipeTokenAccount(address,address,int64)
wipeTokenAccountNFT(address,address,int64[])

### PR DESCRIPTION
**Pull request by lukelee-sl**
**Description**:
Validate that giving incorrect numeric values when calling system contract functions results in returning a failed call. 

**Related issue(s)**:

Fixes #14312 

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/has/hbarapprove/HbarApproveTranslator.java [hunk 1]
```diff
@@ -104,7 +104,7 @@ private CryptoApproveAllowanceTransactionBody cryptoApproveTransactionBody(
                 .cryptoAllowances(CryptoAllowance.newBuilder()
                         .owner(owner)
                         .spender(operatorId)
-                        .amount(amount.longValue())
+                        .amount(amount.longValueExact())
```
Comment (david-bakin-sl) [inline hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/has/hbarapprove/HbarApproveTranslator.java:107]: ha! I guess that means these tests were a good idea, TY. (And also for the next few files)

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/DispatchForResponseCodeHtsCall.java [hunk 2]
```diff
@@ -219,6 +219,7 @@ public DispatchForResponseCodeHtsCall(
         if (status != SUCCESS) {
             status = failureCustomizer.customize(syntheticBody, status, enhancement);
             recordBuilder.status(status);
+            return reversionWith(gasRequirement, recordBuilder);
```
Comment (david-bakin-sl) [inline hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/DispatchForResponseCodeHtsCall.java:222]: yikes.

**Review by david-bakin-sl — APPROVED** [review]
LGTM!!

**Review by MrValioBg — APPROVED** [review]
LGTM!

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/has/hbarapprove/HbarApproveTranslator.java (hunk 1 above)
Comment (lukelee-sl) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/has/hbarapprove/HbarApproveTranslator.java:107]: agreed

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/DispatchForResponseCodeHtsCall.java (hunk 2 above)
Comment (lukelee-sl) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/DispatchForResponseCodeHtsCall.java]: According to @tinker-michaelj , the expected behavior is to not revert.  Took that change out.

Comment (david-bakin-sl) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/DispatchForResponseCodeHtsCall.java]: Ummm, OK if he says so.  But what's the rule?

**Review by david-bakin-sl — APPROVED** [review]
LGTM!  (But see my question above, on a previous comment.  Curious about it.)

**Pull request merged**